src/polyswarm_api/__init__.py

```diff
@@ -337,40 +385,74 @@ def _fix_result(self, result):
         :param filename: Filename to use
         :return: JSON report
         """
-
-        result = await self.post_file(to_scan, filename)
-        if result['status'] == 'OK':
-            uuid = result['result']
-        else:
-            logger.error('Failed to get UUID for scan of file %s', filename)
-            return {'filename': filename, 'files': []}
-
-        logger.info('Successfully submitted file %s, UUID %s', filename, uuid)
-
-        return await self._wait_for_uuid(uuid)
+        return await self.scan_fileobjs([(to_scan, filename)], artifact_type=ArtifactType.FILE)[0]
 
     async def scan_data(self, data, filename=None):
         """
         Scan bytes using the PS API asynchronously.
-
         :param data: Data (in bytes) to submit to be scanned
         :param filename: Filename to use in submission
         :return: JSON report
         """
         if filename is None:
             filename = hashlib.sha256(data).hexdigest()
-        return await self.scan_fileobj(io.BytesIO(data), filename)
+        return await self.scan_fileobjs([(io.BytesIO(data), filename)], artifact_type=ArtifactType.FILE)[0]
+
+    async def scan_url(self, to_scan, name='url'):
+        """
+        Scan a single URL using the PS API synchronously
+
+        :param to_scan: URL to scan
+        :param name: name to associate with the artifact
+        :return: JSON report
+        """
+        return await self.scan_urls([to_scan])[0]
 
     async def scan_file(self, to_scan):
         """
-        Scan a single file using the PS API asynchronously.
+        Scan a single file using the PS API synchronously.
```

Comment:
Hrm I think all of these should be "asynchronously"; I know the particular task waits for completion but does it in an async task.

Reply:
Woops, bad copy/paste.